frontend/src/tw-components/Buttons.tsx

```diff
@@ -0,0 +1,100 @@
+import React from "react";
+import Typography from "tw-components/Typography";
+
+// size can only be one of:
+// "small" | "small-long" | "medium" | "medium-long" | "large" | "large-long" | "icon-only"
+type ButtonSize = keyof typeof buttonSizes;
+// state can only be "enabled" or "disabled"
+type ButtonState = keyof typeof buttonColors;
+
+type ButtonProps = {
+  size?: ButtonSize;
+  state?: ButtonState;
+  children?: React.ReactNode;
+  onClick?: () => void;
+};
+
+type IconButtonProps = {
+  state?: ButtonState;
+  onClick?: () => void;
+};
+
+const buttonSizes = {
+  small: "px-[24px] h-[32px] flex items-center justify-center leading-none",
+  "small-long":
+    "px-[40px] h-[32px] flex items-center justify-center leading-none",
+  medium: "px-[32px] h-[42px] flex items-center justify-center leading-none",
+  "medium-long":
+    "px-[48px] h-[42px] flex items-center justify-center leading-none",
+  large: "px-[40px] h-[51px] flex items-center justify-center leading-none",
+  "large-long":
+    "px-[56px] h-[51px] flex items-center justify-center leading-none",
+  "icon-only": "px-[24px] h-[42px] flex items-center justify-center",
+};
+
+const buttonColors = {
+  enabled:
+    "bg-blue-dark text-white dark:bg-white dark:text-blue-dark hover:bg-blue-dark-hover focus:bg-blue-dark-focused focus:outline-none active:bg-blue-dark-focused dark:hover:bg-grey-light dark:focus:bg-[#D9DBDF] dark:active:bg-[#D9DBDF]",
+  disabled:
+    "bg-grey text-white cursor-not-allowed dark:bg-grey dark:text-grey-light",
+};
+
+const buttonTypography: Record<keyof typeof buttonSizes, React.ElementType> = {
+  small: Typography.Title7,
+  "small-long": Typography.Title7,
+  medium: Typography.Title6,
+  "medium-long": Typography.Title6,
+  large: Typography.Title5,
+  "large-long": Typography.Title5,
+  "icon-only": Typography.Title7, // Not sure about this one
+};
+
+const Button: React.FC<ButtonProps> = ({
+  size = "medium",
+  state = "enabled",
```

Comment:
```suggestion
  disabled = false,
```
Dont pass a prop called `state`, the name is too similar to react's `useState` and component state concept.

Since it can only be either `"enabled"` or `"disabled"`, we can pass a prop called `disabled` whose default value is `false`. What do you think?

Reply:
It's a great idea! I'll make the changes.